### PR TITLE
Make validation errors less verbose

### PR DESCRIPTION
**Pull request by JosephMarinier**
## Description:

Here is an example of a response to a `PATCH` `/config` with an invalid config:

- Before:
> 1 validation error for AzimuthConfig
uncertainty -> iterations
  ensure this value is greater than or equal to 1 (type=value_error.number.not_ge; limit_value=1)

- After:
> AzimuthConfig['uncertainty']['iterations']: ensure this value is greater than or equal to 1

And here is an extreme example with 4 errors (`a`, `b`, `c` and `d`) in a `GET` `/dataset_splits/c/utterances?outcome=a&outcome=b&sort=d&pipeline_index=0`

- Before:
> 5 validation errors for Request
query -> outcome -> 0
  value is not a valid enumeration member; permitted: 'CorrectAndPredicted', 'CorrectAndRejected', 'IncorrectAndPredicted', 'IncorrectAndRejected' (type=type_error.enum; enum_values=[<OutcomeName.CorrectAndPredicted: 'CorrectAndPredicted'>, <OutcomeName.CorrectAndRejected: 'CorrectAndRejected'>, <OutcomeName.IncorrectAndPredicted: 'IncorrectAndPredicted'>, <OutcomeName.IncorrectAndRejected: 'IncorrectAndRejected'>])
query -> outcome -> 1
  value is not a valid enumeration member; permitted: 'CorrectAndPredicted', 'CorrectAndRejected', 'IncorrectAndPredicted', 'IncorrectAndRejected' (type=type_error.enum; enum_values=[<OutcomeName.CorrectAndPredicted: 'CorrectAndPredicted'>, <OutcomeName.CorrectAndRejected: 'CorrectAndRejected'>, <OutcomeName.IncorrectAndPredicted: 'IncorrectAndPredicted'>, <OutcomeName.IncorrectAndRejected: 'IncorrectAndRejected'>])
path -> dataset_split_name
  value is not a valid enumeration member; permitted: 'eval', 'train', 'all' (type=type_error.enum; enum_values=[<DatasetSplitName.eval: 'eval'>, <DatasetSplitName.train: 'train'>, <DatasetSplitName.all: 'all'>])
path -> dataset_split_name
  value is not a valid enumeration member; permitted: 'eval', 'train', 'all' (type=type_error.enum; enum_values=[<DatasetSplitName.eval: 'eval'>, <DatasetSplitName.train: 'train'>, <DatasetSplitName.all: 'all'>])
query -> sort
  value is not a valid enumeration member; permitted: 'index', 'utterance', 'label', 'prediction', 'confidence' (type=type_error.enum; enum_values=[<UtterancesSortableColumn.index: 'index'>, <UtterancesSortableColumn.utterance: 'utterance'>, <UtterancesSortableColumn.label: 'label'>, <UtterancesSortableColumn.prediction: 'prediction'>, <UtterancesSortableColumn.confidence: 'confidence'>])

- After:
> query parameter outcome=a: value is not a valid enumeration member; permitted: 'CorrectAndPredicted', 'CorrectAndRejected', 'IncorrectAndPredicted', 'IncorrectAndRejected'
query parameter outcome=b: value is not a valid enumeration member; permitted: 'CorrectAndPredicted', 'CorrectAndRejected', 'IncorrectAndPredicted', 'IncorrectAndRejected'
path parameter dataset_split_name=c: value is not a valid enumeration member; permitted: 'eval', 'train', 'all'
query parameter sort=d: value is not a valid enumeration member; permitted: 'index', 'utterance', 'label', 'prediction', 'confidence'

## Checklist:

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

**Comment by JosephMarinier** [comment]
I don't understand the problem with the documentation test failing. It has hardly anything to do with this PR, so I included a simple workaround, and we should look into it in another PR. Ready for review!

File: azimuth/app.py [hunk 1]
```diff
@@ -61,12 +61,41 @@ class HTTPExceptionModel(BaseModel):
     detail: str
 
 
-async def handle_validation_error(request: Request, exception: RequestValidationError):
+async def handle_validation_error(request: Request, exception: ValidationError):
```
Comment (gabegma) [inline azimuth/app.py:64]: I think it could be good at some point to move these functions outside of this file? It’s becoming quite big.

**Review by gabegma — APPROVED** [review]
LGTM! 
I don't understand all the details, but thank you for the thorough testing!

It's probably outside the scope of this PR, but when I was playing with it, I put the threshold in the config page to -50%, and got these surprising warnings: 

Screen Shot 2023-04-19 at 21 30 50

I only expected the last one.

File: azimuth/routers/config.py [hunk 2]
```diff
@@ -88,8 +88,10 @@ def patch_config(
         log.error("Rollback config update due to error", exc_info=e)
         new_config = config
         initialize_managers(new_config, task_manager.cluster)
-        if isinstance(e, (AzimuthValidationError, ValidationError)):
+        if isinstance(e, AzimuthValidationError):
             raise HTTPException(HTTP_400_BAD_REQUEST, detail=str(e))
+        if isinstance(e, ValidationError):
+            raise
```
[the remark this answers is not in the document]
Comment (JosephMarinier) [reply azimuth/routers/config.py:94]: Another example in the config would be `use_cuda: Union[Literal["auto"], bool]`.
If we try to `PATCH` the config with
```json
{"use_cuda": "potato"}
```
we get
```
AzimuthConfig['use_cuda']: unexpected value; permitted: 'auto'
AzimuthConfig['use_cuda']: value could not be parsed to a boolean
```
which makes sense.

File: azimuth/app.py (hunk 1 above)
Comment (JosephMarinier) [reply azimuth/app.py:64]: Agree! I'll create `utils/exception_handlers.py`, with the same filename as the default handlers in `site-packages/fastapi/exception_handlers.py`.

File: azimuth/routers/config.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabegma) [reply azimuth/routers/config.py:94]: Thank you for the thorough explanation!!

**Pull request merged**